lib/components/FileAttachment/index.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import "@uppy/core/dist/style.css";
+import "@uppy/progress-bar/dist/style.css";
+import { saveAs } from "file-saver";
+import { isEmpty } from "ramda";
+
+import directUploadsApi from "apis/direct_uploads";
+import { DIRECT_UPLOAD_ENDPOINT } from "common/constants";
+import Button from "components/Common/Button";
+import useUppyUploader from "hooks/useUppyUploader";
+
+import Attachments from "./Attachments";
+import {
+  DEFAULT_UPPY_CONFIG,
+  DROP_DOWN_OPTIONS_KEYS,
+  UPPY_UPLOAD_CONFIG,
+} from "./constant";
+
+import Progress from "../Editor/MediaUploader/Progress";
+
+const FileAttachment = ({
+  endpoint = DIRECT_UPLOAD_ENDPOINT,
+  onRename = _ => {},
+  onDelete = _ => {},
+  onError = _ => {},
+}) => {
+  const fileInputRef = useRef(null);
+  const [uploadedFiles, setUploadedFiles] = useState([]);
+  const [error, setError] = useState("");
+  const uppyConfig = { ...DEFAULT_UPPY_CONFIG };
+  const onClickAttachment = () => fileInputRef?.current?.click();
```

Comment:
```suggestion
  const onClickAttachment = () => fileInputRef.current.click();
```

1. For refs, the `ref.current` part will never need optional chaining.
2. Since this is a single line method with only one occurrence, there is no point of extracting it to a function.